BestPractices.md

```diff
@@ -416,153 +517,368 @@ function Get-Property
 
 **Bad:**
+
 ```powershell
 function Get-MyBoolean
 {
     [OutputType([Boolean])]
     param ()
-    
+
     ...
-    
+
     return $myBoolean
 }
 ```
 
 **Good:**
+
 ```powershell
 function Get-MyBoolean
 {
     [CmdletBinding()]
     [OutputType([Boolean])]
     param ()
-    
+
     ...
-    
+
     return $myBoolean
 }
 ```
 
 ### Return Only One Object From Each Function
 
 **Bad:**
+
 ```powershell
 
 ```
 
 **Good:**
+
 ```powershell
 
 ```
 
 ## DSC Resource Functions
+
 ### Return a Hashtable from Get-TargetResource
 
 **Bad:**
+
 ```powershell
 
 ```
 
 **Good:**
+
 ```powershell
 
 ```
 
 ### Return a Boolean from Test-TargetResource
 
 **Bad:**
+
 ```powershell
 
 ```
 
 **Good:**
+
 ```powershell
 
 ```
 
 ### Avoid Returning Anything From Set-TargetResource
 
 **Bad:**
+
 ```powershell
 
 ```
 
 **Good:**
+
 ```powershell
 
 ```
 
 ### Define Get-TargetResource, Set-TargetResource, and Test-TargetResource for Every DSC Resource
 
 **Bad:**
+
 ```powershell
 
 ```
 
 **Good:**
+
 ```powershell
 
 ```
 
+### Get-TargetResource should not contain unused non-mandatory parameters
+
+**Bad:**
+
+```powershell
+<#
+    .SYNOPSIS
+        Returns the current state of the feature.
+
+    .PARAMETER Name
+        The feature for which to return the state for.
+
+    .PARAMETER ServerName
+        The server name on which the feature is installed.
+
+    .PARAMETER Ensure
+        The desired state of the feature.
+#>
+function Get-TargetResource
+{
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Name,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $ServerName,
+
+        [Parameter()]
+        [System.String]
+        $Ensure
+    )
+
+    Write-Verbose -Message ('{0} for {1}' -f $Name, $ServerName)
+
+    if( $Name )
+    {
+        $feature = 'Enabled'
+    }
+
+    return @{
+        Name = $Name
+        Servername = $ServerName
+        Feeature = $feature
+    }
+}
+```
+
+**Good:**
+
+```powershell
+<#
+    .SYNOPSIS
+        Returns the current state of the feature.
+
+    .PARAMETER Name
+        The feature for which to return the state for.
+
+    .PARAMETER ServerName
+        The server name on which the feature is installed.
+#>
+function Get-TargetResource
+{
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Name,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $ServerName
+    )
+
+    Write-Verbose -Message ('{0} for {1}' -f $Name, $ServerName)
+
+    if( $Name )
+    {
+        $feature = 'Yes'
+    }
+
+    return @{
+        Name = $Name
+        Servername = $ServerName
+        Feeature = $feature
+    }
+}
+```
+
+### Get-TargetResource with unused mandatory parameters should include "Not used in Get-TargetResource" in the comment-based help parameter description
```

Comment:
Can you include a description of why it`s appropriate not to use a mandatory parameter in a get?

For example, in the example you provide Ensure describes a state of the resource which is mandatory, but should not be needed to retrieve the resource.